### PR TITLE
Add simple devcontainer config

### PR DESCRIPTION
**Pull request by spenserblack**
This is just a small configuration to get started using codespaces. Made this while working on another PR :slightly_smiling_face:

### CODE REVIEW

**Comment by Nutlope** [comment]
@spenserblack Can I ask what this helps with? I've never used devcontainers and don't know what they are

**Comment by spenserblack** [comment]
Sure! This sets up a simple dev environment for when contributors want to develop in the cloud environment GitHub Codespaces (I don't think this is limited to Codespaces, but that's how I use it).

Right now the config in this PR is pretty minimal -- it installs node16 and Typescript globally. Then, after the codespace is created, it will automatically run `npm install`. This allows a one-click setup for a dev environment -- just create a codespace for this repo and one is ready to start developing.

Since this config is minimal, I'll admit that it's not necessary. The default codespace image has Node installed, and will automatically run `npm install` or `yarn install`. AFAIK the default image doesn't support pnpm (https://github.com/Nutlope/aicommits/pull/19), so that could be a reason to use a config.

Additionally, if you have any preferred VS Code extensions that you think developers should use (I add `EditorConfig.EditorConfig` to all of my `devcontainer.json`s, for example), this file can be used for that. There could also possibly be a `postAttachCommand` to run a "watch build" every time the user accesses the codespace.

To summarize, this can be used to:
1. Define a container to be used as a dev environment
2. Configure the editor (extensions, setttings, etc.)
3. Run some setup commands on lifecycle hooks

Hope this helps explain!

**Comment by spenserblack** [comment]
Oh, I see now in #19 that the preferred Node version is 18. Since there's a lot of overlap between what goes in `CONTRIBUTING.md` and what goes in `devcontainer.json`, perhaps we should wait until #19 is merged before reviewing this PR further.

**Comment by spenserblack** [comment]
With the recent updates and the switch to using `nvm` and `pnpm`, I'll pick this back up. Closing this PR so that I don't spam you with temp commits.
